sympy/physics/quantum/density.py

```diff
@@ -43,10 +43,12 @@ def _eval_args(cls, args):
 
         for arg in args:
             # Check if arg is a tuple
-            if not (isinstance(arg, Tuple) and
-                    len(arg) == 2):
+            if not (isinstance(arg, Tuple)):
                 raise ValueError("Each argument should be of form [state,prob]"
                                  " or ( state, prob )")
+            elif not len(arg) == 2:
+                raise ValueError("Length should be of size 2") 
```

Comment:
I think the previous check was better. It raises an error with a complete description of the requirement arguments in both cases, rather than leaving the user to guess what the correct argument format is.